Week04/MoviesApp/app/src/main/java/tech/danielwaiguru/moviesapp/repositories/UserRepository.kt

```diff
@@ -0,0 +1,22 @@
+package tech.danielwaiguru.moviesapp.repositories
+
+import tech.danielwaiguru.moviesapp.database.UserDao
+import tech.danielwaiguru.moviesapp.models.User
+
+class UserRepository(private val userDao: UserDao) {
+    /**
+     * Register user
+     */
+    suspend fun registerUser(user: User){
+        userDao.registerUser(user)
+    }
+    fun loginUser(username: String, password: String): Boolean{
+        val user: User? = userDao.loginUser(username)
```

Comment:
this function is a bit weird, as you call "loginUser", but you return a user, and check if the password matches. Maybe rename to `getUserWithName()`